### PR TITLE
adds harm baton to uplink

### PR DESCRIPTION
**Pull request by 9percents**
Harm Baton. Looks similar to stun baton, but instead of stun, it inflicts pain. It also deals increased damage when enabled(on harm intent). Has a dial that sets its power from 1 to 10. Dial increases damage, pain dealt and cell charge cost. Cell charge cost increases exponentially.

Costs 12 TC, 9 TC for sec jobs.
harmbaton_showcase_32x32

:cl:
 * rscadd: Adds a harm baton to uplink. Instead of stun, it deals pain. Its damage is increased when it's enabled.  Security jobs get a discount.

### CODE REVIEW

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -597,6 +597,14 @@ var/list/uplink_items = list()
 	discounted_cost = 8
 	jobs_with_discount = list("Security Officer", "Warden", "Head of Security")
 
+/datum/uplink_item/jobspecific/command_security/syndibaton
+	name = "Harm Baton"
+	desc = "A stun baton modified with tesla relay coils capable of discharing high amount of shock to overload human pain registers. It can also use this energy to boost the impact of baton."
```
Comment (boy2mantwicethefam) [inline code/datums/uplink_item.dm]: discharging

File: code/game/objects/items/weapons/stunbaton.dm [hunk 2]
```diff
@@ -270,3 +270,178 @@
...
+				"<span class='warning'>[src] has no pulse and its soul has departed...</span>")
+		else if (bcell.maxcharge < hitcost)
+			to_chat(user, "<span class='warning'>[src] clicks but nothing happens. Something must be wrong with the battery.</span>")
+		else
+			user.simple_message("<span class='warning'>[src] is out of charge.</span>",
+				"<span class='warning'>[src] refuses to obey you.</span>")
+
+	add_fingerprint(user)
+
+/obj/item/weapon/melee/baton/harm/attack(mob/M, mob/user)
```
Comment (boy2mantwicethefam) [inline code/game/objects/items/weapons/stunbaton.dm]: >copying the entire proc

*scream

**Comment by dunsel-hordon** [comment]
can you pop it open and change the battery?

**Comment by 9percents** [comment]
@dunsel-hordon yes, 3rd sprite on the showcase is baton without its cell

File: code/game/objects/items/weapons/stunbaton.dm (hunk 2 above)
Comment (9percents) [reply code/game/objects/items/weapons/stunbaton.dm]: eh, not much I can do, I placed ..() wherever I could to make thinner

**Review by Kurfursten — CHANGES_REQUESTED** [review]
Do this in a more OOP way, there's way too much pastecode here. It's a child of batons, so you shouldn't need to copy the bulk of what you have here. So many of these behaviors are identical - logging, visible messages, sound on hit, deducting charge, etc. -- all of this should just be inherited from the parent.

**Comment by boy2mantwicethefam** [comment]
Yea, try adding a bunch of variables to the parent that are changed in the child

**Comment by 9percents** [comment]
I'll resolve conflicts soon

File: code/game/objects/items/weapons/stunbaton.dm [hunk 3]
```diff
@@ -127,8 +131,13 @@
 	if(status && clumsy_check(user) && prob(50))
 		user.simple_message("<span class='warning'>You grab the [src] on the wrong side.</span>",
 			"<span class='danger'>The [name] blasts you with its power!</span>")
-		user.Knockdown(stunforce)
-		user.Stun(stunforce)
+		if(istype(src, /obj/item/weapon/melee/baton/harm))
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/stunbaton.dm]: this isn't OOP!

there are 2 better ways to do this
1. Just have a var it checks, and set the var to something different on your new harmbaton; e.g.: `if(hittype == BATON_AGONY)`...
2. Create a `baton_hit(mob/victim)` proc and call that - have normal behavior in the baton and then override it in your harm baton

**Review by Kurfursten — CHANGES_REQUESTED** [review]

**Review by Kurfursten — APPROVED** [review]
This is good

**Pull request merged**